### PR TITLE
Bulk upload implementataion

### PR DESCRIPTION
**Pull request by VickyRavi**

### CODE REVIEW

File: src/main/java/com/sellinall/jdlisting/validation/ValidateMandatoryFields.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+			String warrantyPeriod = jdObject.getString("warrantyPeriod");
+			String warrantyPeriodCode = JDUtil.warrantyPeriod.get(warrantyPeriod.toLowerCase().replaceAll(" ", ""));
+			exchange.setProperty("warrantyPeriodCode", warrantyPeriodCode);
+			if (warrantyPeriodCode == null) {
+				failureReason += "# Invalid warranty period \n";
+			}
+		}
+		if (jdObject.get("isCod") == null) {
+			failureReason += "# Please give isCod(Y/N)\n";
+		}
```
Comment (VickyRavi) [inline src/main/java/com/sellinall/jdlisting/validation/ValidateMandatoryFields.java]: validate COD

File: src/main/java/com/sellinall/jdlisting/validation/ValidateMandatoryFields.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		if (jdObject.get("imageURI") != null) {
+			BasicDBList imageURI = (BasicDBList) jdObject.get("imageURI");
+			if (imageURI.size() > 6) {
+				failureReason += "# Upto six images can be given \n";
+			} else if (imageURI.size() == 0) {
+				failureReason += "# Please give item image \n";
+			}
+		}
+		if (jdObject.get("tags") != null) {
+			BasicDBList tags = (BasicDBList) jdObject.get("tags");
```
Comment (VickyRavi) [inline src/main/java/com/sellinall/jdlisting/validation/ValidateMandatoryFields.java]: check size here itself

File: src/main/java/com/sellinall/jdlisting/validation/ValidateMandatoryFields.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+			if (sku.contains("-")) {
+				BasicDBList childJd = ((BasicDBList) inventory.get("jd"));
+				BasicDBObject childJdObject = (BasicDBObject) childJd.get(0);
+				if (childJdObject.get("imageURI") != null) {
+					BasicDBList imageURI = (BasicDBList) childJdObject.get("imageURI");
+					if (imageURI.size() > 6) {
+						failureReason += "# Upto six images can be given child item\n";
+					} else if (imageURI.size() == 0) {
+						failureReason += "# Please give child item image \n";
+					}
```
Comment (VickyRavi) [inline src/main/java/com/sellinall/jdlisting/validation/ValidateMandatoryFields.java]: remove child item check

File: src/main/java/com/sellinall/jdlisting/validation/ValidateMandatoryFields.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+				}
+			}
+		}
+
+		if (!failureReason.isEmpty()) {
+			exchange.setProperty("isPostingSuccess", false);
+			exchange.setProperty("isUpdateSuccess", false);
+			exchange.setProperty("isFieldsValidationSuccess", false);
```
Comment (VickyRavi) [inline src/main/java/com/sellinall/jdlisting/validation/ValidateMandatoryFields.java]: remove unused property

File: src/main/java/com/sellinall/jdlisting/request/SubmitAddItemAPI.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+					} else {
+						JSONObject model = openapiData.getJSONObject("model");
+						String countryCode = exchange.getProperty("countryCode", String.class);
+						BasicDBObject parentInventory = exchange.getProperty("parentInventory", BasicDBObject.class);
+						long spuId = model.getLong("spuId");
+						String itemTitle = parentInventory.getString("itemTitle").replaceAll(" ", "-");
+						exchange.setProperty("spuId", spuId);
+						exchange.setProperty("itemUrl",
+								Config.getItemUrl(countryCode) + itemTitle + "_" + spuId + ".html");
+						exchange.setProperty("failureReason", "");
```
Comment (VickyRavi) [inline src/main/java/com/sellinall/jdlisting/request/SubmitAddItemAPI.java]: remove url

File: src/main/java/com/sellinall/jdlisting/request/SubmitUploadImages.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+			throws Exception {
+		String[] imageNameContent = imageName.split("/");
+		JSONObject payload = new JSONObject();
+		payload.put(paramName, id);
+		payload.put("fileName", imageNameContent[imageNameContent.length - 1]);
+		JSONObject response = CallApiUtil.uploadImage(apiName, accessToken, clientId, appSecret, payload, imageName,
+				"ID");
+		if (response.has("httpCode") && response.getInt("httpCode") != HttpStatus.OK_200) {
+			log.error("Error occured while uploading images for sku " + SKU);
+		} else {
```
Comment (VickyRavi) [inline src/main/java/com/sellinall/jdlisting/request/SubmitUploadImages.java]: print response

File: src/main/resources/CamelContextPost.xml [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+					<when>
+						<simple>${exchangeProperty.isFieldsValidationSuccess} == true
+						</simple>
+						<to uri="direct:invokeAddItemAPI" />
+					</when>
+				</when>
+				<doCatch>
+					<exception>java.lang.Exception</exception>
+					<setHeader headerName="exceptionMessage">
+						<simple>${exception.stacktrace}</simple>
```
Comment (VickyRavi) [inline src/main/resources/CamelContextPost.xml]: posting exception

File: src/main/java/com/sellinall/jdlisting/db/PrepareItemStatusUpdateDetails.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+			log.error("Error occured while updating inventory status for the SKU= "
+					+ exchange.getProperty("SKU", String.class));
+			updateInventoryList = new ArrayList<BasicDBObject>();
+			updateInventory = new BasicDBObject();
+			updateInventory.put("SKU", Pattern.compile(SKU.split("-")[0] + ".*"));
+			updateInventory.put("failureReason", "InternalError");
+			updateInventory.put("elevenStreet.$.status", SIAInventoryStatus.FAILED.toString());
+			updateInventoryList.add(updateInventory);
+			exchange.setProperty("failureReason", "Internal error");
+		}
```
Comment (VickyRavi) [inline src/main/java/com/sellinall/jdlisting/db/PrepareItemStatusUpdateDetails.java]: remove

File: src/main/java/com/sellinall/config/Config.java [hunk 9]
```diff
@@ -16,6 +16,7 @@
 	private String jdApiBaseUrl;
```
Comment (Deepakelango) [inline src/main/java/com/sellinall/config/Config.java]: Reviewed

**Pull request merged**